### ISSUE TITLE
Documentation on readthedocs

### ISSUE THREAD

**Issue by glennmatthews** [networktocode/diffsync#33]
### Environment
* DiffSync version:  1.0.0

### Proposed Functionality

Documentation generation and publication to readthedocs.org, including examples

### Use Case

Discoverability, usability.

**Comment by itdependsnetworks** [comment]
Would be great to include a few examples in the docs as well.

**Comment by glennmatthews** [comment]
Agreed. We have the one example (`examples/example1/`) but more would definitely be desirable.

**Comment by glennmatthews** [comment]
It looks like Poetry + ReadTheDocs is not a trivially solved problem - see https://github.com/readthedocs/readthedocs.org/issues/4912, although there are some possible workarounds listed there.

**Comment by dgarros** [comment]
@josh5276 please can you look into this issue
We should be able to replicate the work that has been done in https://github.com/networktocode/netutils
The goal of this issue is to integrate this repo with RTD and move what we have in the README into a basic doc site for now
I'll open more issue to improve the documentation

**Comment by josh5276** [comment]
Sure thing @dgarros, I'll look into it.
